### PR TITLE
Editable custom fields #21

### PR DESCRIPTION
**Pull request by oconnor-sn**
Change made as per Issue #21.
Code changed in _send_signature_request_with_template

(shared logic between send_signature_request_with_template
            and send_signature_request_embedded_with_template)

Please review @martinytodorov and revert with any issues.

### CODE REVIEW

File: hellosign_sdk/hsclient.py [hunk 1]
```diff
@@ -2116,7 +2116,8 @@ def _send_signature_request_with_template(self, test_mode=False, client_id=None,
             "message": message,
             "signing_redirect_url": signing_redirect_url,
             "allow_decline": self._boolean(allow_decline),
-            "signing_options": HSFormat.format_json_data(signing_options)
+            "signing_options": HSFormat.format_json_data(signing_options),
+            "custom_fields": HSFormat.format_json_data(custom_fields)
```
Comment (martinytodorov) [inline hellosign_sdk/hsclient.py:2120]: @oconnor-sn adding custom fields here will not solve the problem. In fact the custom fields are being added to the request `data` (see line 2134).

To make custom fields support `name, value, editor, required` you'd need to look into this function:
`format_custom_fields` defined in `hsformat.py`.

Please share with me the python example you have when testing this. I want to make sure you have a good request example so you can correctly test this. Feel free to send it over slack or here and remember to not include your api key in it.

**Review by martinytodorov — CHANGES_REQUESTED** [review]

File: hellosign_sdk/hsclient.py (hunk 1 above)
Comment (oconnor-sn) [reply hellosign_sdk/hsclient.py:2120]: thanks @martinytodorov - I will send via slack.

Comment (martinytodorov) [reply hellosign_sdk/hsclient.py:2120]: Hey @oconnor-sn, here's the problem:
We have customers that are using the custom fields with their old syntax. If you see the function I am pointing you at (format_custom_fields) you will see that the expected format is `custom_fields=[{'fieldid1':'somevalue1', 'fieldid2':'somevalue2'}]`

Your fix should not be introducing a new breaking change. See your example changed here for 2 roles and 2 types of custom fields syntaxes. Both syntaxes should work:
https://gist.github.com/martinytodorov/d3bf0021d7d3e1952811fcedf44b6046

**Comment by jtreminio-dropbox** [comment]
Please use the new OpenAPI SDK: https://pypi.org/project/dropbox-sign/

**Pull request closed without merging**